File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let indexMap = new Map();
    
    // Iterate through the 'nums' array and store the indices in the map
    for (let i = 0; i < nums.length; i++) {
      indexMap.set(nums[i], i + 1);
    }

    // Create an empty array to store the result
    let result = [];

    // Iterate through the 'nums' array again to find the pair that adds up to the 'target'
    for (let i = 0; i < nums.length; i++) {
      let complement = target - nums[i];
      let j = indexMap.get(complement) - 1;

      // Check if the complement exists in the map and the indices are not the same
      if (j >= 0 && j !== i) {
        // Assign the minimum and maximum of the indices to the result array
        result = [Math.min(i, j), Math.max(i, j)];
        return result;
      }
    }

    // If no solution is found, return an empty array
    return result;

};

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int>x; 
        int n = nums.size(); 
        for(int i = 0; i < n ; i++){
            x[nums[i]] = i+1;
        }
        vector<int>a; 
        for(int i = 0; i < n ; i++){
            int j = x[target - nums[i]]-1 ;
            if( j >=0 && j!= i  ){
                a = {min(i , j ) , max(i , j ) }; 
                return a ; 
            }
        }
        return a; 
    }
};